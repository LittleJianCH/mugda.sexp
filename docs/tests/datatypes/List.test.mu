(import "List.mu" List null cons length)
(import "Nat.mu" Nat zero)

(null Nat)
(cons Nat zero (null Nat))
(cons Nat zero (cons Nat zero (null Nat)))

(length Nat (cons Nat zero (cons Nat zero (null Nat))))
