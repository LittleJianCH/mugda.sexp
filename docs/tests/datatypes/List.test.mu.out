(null Nat)
(cons Nat zero (null Nat))
(cons Nat zero (cons Nat zero (null Nat)))
((fn [(A (null A)) zero] [(A (cons A head tail)) (add1 (length A tail))]) Nat (cons Nat zero (cons Nat zero (null Nat))))
